internal/hcsoci/resources_lcow.go

```diff
@@ -6,18 +6,39 @@ package hcsoci
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"os"
 	"path"
 	"path/filepath"
 	"strings"
 
 	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/oci"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
 )
 
 const lcowMountPathPrefix = "/mounts/m%d"
+const lcowGlobalMountPrefix = "/run/mounts/m%d"
+
+// keep lcowNvidiaMountPath value in sync with opengcs
+const lcowNvidiaMountPath = "/run/nvidia"
+
+// getGPUVHDPath gets the gpu vhd path from the shim options or uses the default if no
+// shim option is set. Right now we only support Nvidia gpus, so this will default to
+// a gpu vhd with nvidia files
+func getGPUVHDPath(coi *createOptionsInternal) (string, error) {
+	gpuVHDPath, ok := coi.Spec.Annotations[oci.AnnotationGPUVHDPath]
+	if !ok || gpuVHDPath == "" {
+		// default path was not set in config file, switch to default
+		gpuVHDPath = filepath.Join(filepath.Dir(os.Args[0]), "nvidiagpu.vhd")
```

Comment:
I dont like this. If its not in the spec (config) then we fail with "no gpu.vhd". I dont like auto upgrading based on where the binary is because it is prone to path issues and also it doesnt consider the LinuxBootFiles dir where we store the rest of the lcow stack.